src/main/java/com/github/seikichi/pronto/infer/example/App.java

```diff
@@ -10,4 +10,22 @@ public String getGreeting() {
   public static void main(String[] args) {
     System.out.println(new App().getGreeting());
   }
+
+  public int test1() {
+    String s = null;
+    return s.length();
+  }
+
+  @ThreadSafe
+  public static class Dinner {
+    private int mTemperature;
+
+    public void makeDinner() {
+      boilWater();
```

Comment:
Unprotected write. Non-private method `void App$Dinner.makeDinner()` indirectly writes to field `this.mTemperature` outside of synchronization.
 Reporting because the current class is annotated `@ThreadSafe`, so we assume that this method can run in parallel with other non-private methods in the class (including itself).